src/Courses/CourseDetailsTabs/TopicsTab.tsx

```diff
@@ -367,18 +437,50 @@ export const TopicsTab: React.FC<TopicsTabProps> = ({ course, setCourse }) => {
                                             // Grabbing this for error handling (see default below)
                                             const { status } = event;
                                             if (status !== 'error') {
-                                                setError(null);
+                                                setAlert({
+                                                    variant: 'info',
+                                                    message: ''
+                                                });
                                             }
                                             if (status !== 'loading') {
                                                 setLoading(false);
                                             }
                                             switch (event.status) {
                                             case 'error':
-                                                setError(event.data);
+                                                if (event.data instanceof BackendAPIError) {
+                                                    const data = event.data.data as {
+                                                        missingPGFileErrors: Array<string>;
+                                                        missingAssetFileErrors: Array<string>;                                                    
+                                                    };
+                                                    setAlert({
+                                                        variant: 'danger',
+                                                        message: CourseTarballImportWarnings({
+                                                            message: 'The course archive upload failed with the following errors:',
+                                                            missingAssetFileErrors: data.missingAssetFileErrors,
+                                                            missingPGFileErrors: data.missingPGFileErrors,
+                                                        })
+                                                    });
+                                                } else {
+                                                    setAlert({
+                                                        variant: 'danger',
+                                                        message: event.data.message
+                                                    });
+                                                }
                                                 break;
-                                            case 'success':
+                                            case 'success': {
```

Comment:
Do we not set a success modal?

Reply:
No because the unit comes up, this matches the existing add unit behavior